examples/transactions/transactional_consumer.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+func startConsuming(ctx context.Context, kafkaBrokers string, topic string) {
+	client, err := kgo.NewClient(
+		kgo.SeedBrokers(strings.Split(kafkaBrokers, ",")...),
+		// Only read messages that have been written as part of committed transactions.
+		kgo.FetchIsolationLevel(kgo.ReadCommitted()),
+	)
+	if err != nil {
+		fmt.Printf("error initializing Kafka consumer: %v\n", err)
+		return
+	}
+
+	client.AssignGroupTransactSession(
+		"my-consumer-group",
+		kgo.GroupTopics(topic),
+	)
```

Comment:
You likely want to `defer client.AssignGroup("")` to leave the group, rather than shutting down without leaving (which leaves the responsibility to Kafka to eventually notice the heartbeat interval has lapsed and the group member is missing).

Reply:
As one potential suggestion/point of feedback, would it be worth considering adding a helper method like `client.LeaveGroup()` that does this call with the empty string?

Reply:
With Sarama, you have to call `Close` on the `ConsumerGroup` interface you have. I'm open to adding a `LeaveGroup()`. I'm not 100% satisfied with the current commit offsets functions, so there's definitely room for improvement in the broader API as a whole (so if you have even more ideas I'm open to them). Unfortunately this would likely require a breaking change in the future if a much better API is found (before stabilizing on 1.0.0)

Reply:
Actually, I've remembered that `client.Close()` leaves the group as well (unless you are using instance IDs). Do you think an explicit `LeaveGroup` would still be beneficial?

Reply:
Yep, you're right - from `client.go`:
```
// Close leaves any group and closes all connections and goroutines.
func (cl *Client) Close() {
```
It would be nice to have if the expected way to leave a group was the explicit `client.AssignGroup("")` but given the `Close()` behavior I don't think the `client.LeaveGroup()` is necessary

Reply:
I'm open to adding an explicit function still, it may be helpful.

Reply:
If you don't mind making the API change then I certainly won't argue with adding it. Even though it's _technically_ a little redundant its usage would be clearer and might highlight the need to do this, whether it's via `client.Close()` or the explicit call to `LeaveGroup()` (or whatever)

Reply:
FWIW I've added `LeaveGroup`, it's on the latest untagged commits -- but I think your update to use `client.Close` is great here.

Reply:
(forgot to push, now it's pushed -- I plan to tag tomorrow or the next day)

Reply:
I've cut v0.6.4 after running race tests for a bit -- turns out that there were a few small bugs, two that were tagged, one not yet tagged.